### ISSUE TITLE
multiple kinects not working

### ISSUE THREAD

**Issue by naychrist** [gameoverhack/ofxOpenNI#11]
running the example with the experimental branch it looks like kinects are detected and initialised okay but the second kinect output is always rendered black.

**Comment by gameoverhack** [comment]
hi there...sorry I haven't got to this sooner...i just checked this on OSX and all looks fine...what platform are you using?

**Comment by naychrist** [comment]
Hey no worries. I am on OSX 10.6.8 but am now suspecting this to be a 
USB bus issue. I have ordered a couple of USB IDE cards to install and 
see if this is the case but haven't had a chance yet as I have a 
deadline this week. I will try and get to this early next week and post 
back.

**Comment by chrisporter** [comment]
Yeah you need the Kinects in root hubs. You can get a StarTech 4 independent port USB PCI card but I usually find most PCs anyway have at least 2 root hubs.

BTW how do you use multiple Kinects anyway ? There's no option to set device ID ?

**Comment by ghost** [comment]
Hi,

I am trying out the experimental ofxOpenNI but I'm having some problems ....
How can I have the hands drawn on the screen? How are hands handled in the new version?
I want to switch between live video and player a lot of times, and some times the thread stops, and when I close the application it crashes with xnWaitAnyUpdateAll().... how can I avoid this kind of problem?

**Comment by gameoverhack** [comment]
hi ptrigueiros that is a forum question rather than a response or addition to this issue - which is specifically about getting multiple Kinects to work. Could you repost to http://forum.openframeworks.cc/index.php/topic,7403.165.html ?

**Comment by kylemcdonald** [comment]
sorry, i realize this is kind of an old issue :) but, similar to @chrisporter it's not clear to me how to access multiple kinects? more importantly, it's not clear if it's possible to do skeleton tracking on multiple kinects? i've read some things that say openni 2.0 supports this but not earlier versions... any hints towards this end would be appreciated :)

**Comment by timmb** [comment]
Independent skeleton tracking from two kinects on a single machine has always been a holy grail for me!

As far as I've seen, there is currently no way to get OpenNI 2.0 working with Kinect though (although I've not tried it personally).

Check out this thread though http://openni-discussions.979934.n3.nabble.com/OpenNI-dev-Multiple-Skeleton-Tracking-td4007206.html , which suggests that you can get multiple skeleton tracking but only by running two separate programs.

**Comment by kylemcdonald** [comment]
wow, super helpful discussion thread. i should have done a more general search first :) it sounds like, at the moment, if i have two processes (apps) separately connected to the two kinects, it's definitely possible. but i still don't see a way of specifying which device i'm connected to in ofxOpenNI. i'll dig around a bit more.

the most recent response on that thread makes it sound like maybe OpenNI2 supports multi-user skeleton tracking.

**Comment by kylemcdonald** [comment]
sorry, it looks like all you have to do to access multiple kinects is just create multiple ofxOpenNI objects. but it's not possible right now to have multiple kinects doing skeleton tracking in the same process. however, you can do it if you build separate apps using a simple `setDeviceID()` method i just added: https://github.com/gameoverhack/ofxOpenNI/pull/30

maybe openni2 supports multiple skeletons in the same process, but afaict openni2 only runs on linux and windows right now.
